### ISSUE TITLE
FATAL ERROR: DNS looks broken for <domain>

### ISSUE THREAD

**Issue by namontague** [omega8cc/boa#1369]
In a clean Debian 9 AMI from AWS the install fails to complete.

server.domain.com is a placeholder
email@email.com is a placeholder
131.101.111.111 is a placeholder

**Command:**
`boa in-head public server.domain.com email@email.com o1 ask php-7.1`

**Output:**

-------------------------------------------------------------------------------------------------------------------
BOA [15:47:07] ==> Skynet Agent v.BOA-4.0.0-prod welcomes you aboard!

BOA [15:47:10] ==> INFO: NORMAL INIT
BOA [15:47:10] ==> INFO: Public Setup Mode Active
BOA [15:47:11] ==> INFO: Creating your /root/.barracuda.cnf config file
BOA [15:47:16] ==> VIRT: This system is supported: Xen guest fully virtualized (HVM)

BOA [15:47:16] ==> Aegir on ip-172-31-18-16 Debian/stretch x86_64 PHP 7.1

BOA [15:47:16] ==> INFO: Installing some basic tools now, please wait...
BOA [15:47:28] ==> INFO: Installing sysvinit-core on Debian stretch...

BOA [15:47:39] ==> NOTE: Please reboot and run barracuda upgrade
BOA [15:47:39] ==> NOTE: when boa will complete initial installation
BOA [15:47:39] ==> NOTE: to cleanly remove not used systemd packages!

BOA [15:47:55] ==> INFO: Installing more basic tools now, please wait...
BOA [15:48:55] ==> INFO: DNS test: OK

BOA [15:48:55] ==> INSTALL START -> checkpoint:

    * Your email address is email@email.com
    * Your IP address is 131.101.111.111
    * Your hostname is server.domain.com

BOA [15:48:55] ==> INFO: Installing extra Drush versions...
BOA [15:49:01] ==> INFO: Running system packages update...
BOA [15:50:21] ==> INFO: Installing required libraries and tools, please wait...
BOA [15:52:22] ==> INFO: Installed OpenSSH version 7.4p1, upgrade required
BOA [15:52:22] ==> INFO: Building OpenSSH 7.9p1 from sources, please wait...
BOA [15:55:13] ==> INFO: Installing MariaDB...
BOA [15:55:26] ==> INFO: Retrieving F1656F24C74CD1D8 key..
BOA [15:55:30] ==> INFO: Retrieving 9334A25F8507EFA5 key..
BOA [15:55:32] ==> INFO: Installing MariaDB 10.1.38 in Debian/stretch
BOA [15:57:17] ==> INFO: Running aptitude full-upgrade...
BOA [16:00:15] ==> INFO: Installed Nginx version nginx/1.10.3, upgrade required
BOA [16:00:15] ==> INFO: Installing Nginx...
BOA [16:02:48] ==> INFO: Installing MySecureShell master-20-03-2015...
BOA [16:04:30] ==> INFO: Installing wkhtmltopdf and wkhtmltoimage x86_64...
BOA [16:04:36] ==> INFO: Fix #1 for libs in Debian stretch
BOA [16:04:36] ==> INFO: Fix #2 for libs in Debian stretch
BOA [16:04:36] ==> INFO: Installing VnStat monitor...
BOA [16:04:52] ==> INFO: Installing a few more tools...
BOA [16:04:52] ==> INFO: PHP 7.1.27 will be installed now
BOA [16:04:52] ==> INFO: Building PHP 7.1.27 from sources, please wait...
BOA [16:04:59] ==> INFO: Building PHP 7.1.27 part 1/3
BOA [16:05:02] ==> INFO: Building PHP 7.1.27 part 2/3
BOA [16:05:42] ==> INFO: Building PHP 7.1.27 part 3/3
BOA [16:16:42] ==> INFO: Installing PhpRedis for PHP 7.1.27...
BOA [16:17:00] ==> INFO: Installing UploadProgress for PHP 7.1.27...
BOA [16:17:07] ==> INFO: Installing JSMin for PHP 7.1.27...
BOA [16:17:14] ==> INFO: Installing Imagick for PHP 7.1.27...
BOA [16:17:31] ==> INFO: Installing LibYAML for PHP...
BOA [16:17:51] ==> INFO: Installing YAML for PHP 7.1.27...
BOA [16:21:02] ==> INFO: Installing Limited Shell 0.9.18.9...
BOA [16:21:17] ==> INFO: Installing Redis update for Debian/stretch...
BOA [16:22:14] ==> INFO: Generating random password for MariaDB
BOA [16:22:14] ==> INFO: Updating MariaDB configuration
BOA [16:22:14] ==> INFO: Restarting MariaDB server...
BOA [16:22:26] ==> INFO: Installing Aegir Master Instance, please wait...
BOA [16:24:32] ==> INFO: Generating DH parameters, 2048 bit, please wait...
BOA [16:24:52] ==> INFO: Aegir Master Instance installed
BOA [16:24:52] ==> INFO: Installing default SSL Wildcard Nginx Proxy...
BOA [16:24:54] ==> INFO: Building Pure-FTPd server from sources, please wait...
BOA [16:24:54] ==> INFO: Generating DH parameters, 2048 bit, please wait...

BOA [16:26:22] ==> INFO: Installing Adminer Manager...
BOA [16:26:23] ==> INFO: Adminer Manager installed
BOA [16:26:23] ==> INFO: Updating init scripts
BOA [16:26:24] ==> INFO: InnoDB log file will be set to 256M...
BOA [16:26:45] ==> INFO: Starting Redis, PHP-FPM and Nginx
BOA [16:26:53] ==> INFO: MariaDB final setup
BOA [16:26:53] ==> INFO: Restarting MariaDB server...
BOA [16:27:04] ==> INFO: MariaDB setup completed
BOA [16:27:04] ==> INFO: You can now log in as root by typing just 'mysql'
BOA [16:27:04] ==> INFO: Cleaning up system swap, it may take a moment...

BOA [16:27:11] ==> INFO: Congratulations, Aegir have been installed successfully!
BOA [16:27:11] ==> NOTE! Please wait 2 min before visiting Aegir at:

BOA [16:27:11] ==> LINK: http://master.server.domain.com/user/reset/1/1555518216/EvGzpWOe7315s9vkAwOS0EPj89i9c1vv3g1poR6VigA/login

BOA [16:27:22] ==> INFO: Installing csf/lfd firewall...
BOA [16:27:26] ==> TEST: csf/lfd firewall should work fine on this system
BOA [16:28:56] ==> INFO: csf/lfd firewall installed
BOA [16:28:56] ==> INFO: Upgrading csf/lfd firewall...
BOA [16:28:59] ==> TEST: csf/lfd firewall should work fine on this system
BOA [16:30:29] ==> INFO: csf/lfd firewall upgrade completed
BOA [16:30:30] ==> INFO: Installing DNS cache pdnsd server...
BOA [16:30:57] ==> INFO: Building pdnsd 1.2.9a from sources, please wait...
curl: (6) Could not resolve host: getcomposer.org
mv: cannot stat 'composer.phar': No such file or directory
BOA [16:31:43] ==> CARD: Now charging your credit card for this magic show...
BOA [16:31:47] ==> CARD: It will take a moment to process your payment...
BOA [16:31:53] ==> JOKE: Just kidding !!! Enjoy your Aegir Hosting System :)

BOA [16:31:57] ==> Final post-install cleaning, one moment...
BOA [16:32:07] ==> BYE!
netcat: getaddrinfo: Temporary failure in name resolution
INFO: The mirror files.aegir.cc doesn't respond, let's try default

BOA [16:32:24] ==> Skynet Agent v.BOA-4.0.0-prod welcomes you aboard!

netcat: getaddrinfo: Temporary failure in name resolution
BOA [16:34:17] ==> WARN: The GitLab mirror repository server doesn't respond...
BOA [16:34:17] ==> INFO: Creating your /root/.o1.octopus.cnf config file
BOA [16:34:18] ==> New Octopus Setup on server.domain.com in progress...

    * Your email address is email@email.com
    * Your client email address is email@email.com
    * Your Aegir control panel for this instance will be available at:
        https://o1.server.domain.com
    * Your Aegir system user for this instance will be o1
    * This Octopus will use PHP-CLI 7.1 for all sites
    * This Octopus will use PHP-FPM 7.1 for all sites
    * This Octopus includes platforms: ALL
    * This Octopus options are listed as SSD / Y / 8 C

BOA [16:42:49] ==> INIT A: Shared platforms code v.001 (new) will be created
BOA [16:42:49] ==> INIT A: Adding user...
BOA [16:42:49] ==> INIT A: FATAL ERROR: DNS looks broken for server.domain.com
BOA [16:42:49] ==> INIT A: FATAL ERROR: Aborting AegirSetupA installer NOW!
BOA [16:42:49] ==> FATAL ERROR: AegirSetupA installer failed
BOA [16:42:49] ==> FATAL ERROR: Aborting Octopus installer NOW!

BOA in-head completed
Bye

-------------------------------------------------------------------------------------------------------------------

Does anyone know why this is failing?

The command` (echo $(getent ahostsv4 server.domain.com) | cut -d: -f2 | awk '{ print $1}' 2>&1)` does return the ip address before the install in run.

**Comment by aanjaneyam** [comment]
I am having a similar problem as above apart form the fact that it fails on on DNS look ups and gives ERROR (not Fatal Error) on brand new Octopus installation . Steps to reproduce:
1) Start a brand new vanilla Debian 9 server on Linode.
2) Configure DNS wild card entry for mydomain.org and check propagation using getent hosts  server.mydomain.org to confirm it is working.
3) Configure SSH (RSA) keys for root.
4) Run the install with command "boa in-head public server.domain.com email@email.com o1"
5) Gives some resolutions errors like in the original post above:
     `a) WARN: The GitLab mirror repository server doesn't respond...` 
      `b) BOA [16:30:57] ==> INFO: Building pdnsd 1.2.9a from sources, please wait...
       curl: (6) Could not resolve host: getcomposer.org
        mv: cannot stat 'composer.phar': No such file or directory`
       `c) netcat: getaddrinfo: Temporary failure in name resolution
        INFO: The mirror files.aegir.cc doesn't respond, let's try default`
        `d) ERROR: 127.0.0.1 doesn't point to server.mydomain.org.`

I then searched for information on how Linode configures networking and name servers for freshly built Debian 9 servers and came across https://www.linode.com/docs/platform/network-helper/ . Linode Network helper configures static networking for Debian 9 linode  on every boot by configuring /etc/network/interfaces and /etc/resolv.conf files.

It seems that the Network Helper correctly configures a static networking for the Linode but the BOA-4.0.0 install (while running Barracuda and Master instance install and Upgrades) overwrites/empties the /etc/resolv.conf thus messing the name resolutions. When I leave the linode running for some time or reboot it the Network helper automatically corrects the networking configuration by configuring /etc/network/interfaces and populating /etc/resolv.conf. However after Network helper's reconfiguration of networking, I run `barracuda up-head` and the barracuda script some where during its execution overwrites/empties the /etc/resolv.conf thus throwing out resolution errors and eventually ending up with a failed install with error.

What is the solution. Should we disable the Linode Network Helper or provide dns-nameservers entry in /etc/network/interfaces as I see that that the package resolvconf is installed or edit some other file.

**Comment by aanjaneyam** [comment]
In fact after running barracuda script (install or upgrade) none of the commands like `hostname -f`, `ping` works until linode helper re-populates the /etc/resolv.conf. Both the commands above return `Temporary failure in name resolution`. The above commands start working automatically after some time until we run barracuda up-head. There it seems like a loop which will never end in successful BOA installation until we correct the problem.

**Comment by aanjaneyam** [comment]
I don't understand what may be going wrong regarding the name resolution issue.
Today I rebuilt Debian 9 with Linode network helper disabled globally and in the server config so that we get vanilla networking and ran the BOA installer again on freshly rebuilt Debian 9.

`root@li761-127:~# boa in-head public server.mydomain.net My@email o1

BOA [14:41:18] ==> Skynet Agent v.BOA-4.0.0-prod welcomes you aboard!

BOA [14:41:21] ==> INFO: NORMAL INIT
BOA [14:41:21] ==> INFO: Public Setup Mode Active
BOA [14:41:22] ==> INFO: Creating your /root/.barracuda.cnf config file
BOA [14:41:26] ==> VIRT: This system is supported: Linux KVM guest

BOA [14:41:26] ==> Aegir on li761-127 Debian/stretch x86_64 PHP 7.2

BOA [14:41:26] ==> INFO: Installing some basic tools now, please wait...
BOA [14:41:38] ==> INFO: Installing sysvinit-core on Debian stretch...

BOA [14:41:43] ==> NOTE: Please reboot and run barracuda upgrade
BOA [14:41:43] ==> NOTE: when boa will complete initial installation
BOA [14:41:43] ==> NOTE: to cleanly remove not used systemd packages!

BOA [14:41:58] ==> INFO: Installing more basic tools now, please wait...
BOA [14:42:26] ==> INFO: DNS test: OK

BOA [14:42:26] ==> INSTALL START -> checkpoint:

    * Your email address is My@email
    * Your IP address is xxx.xxx.xx.xxx
    * Your hostname is server.mydomain.net

BOA [14:42:26] ==> INFO: Installing extra Drush versions...
BOA [14:42:30] ==> INFO: Running system packages update...
BOA [14:43:42] ==> INFO: Installing required libraries and tools, please wait...
BOA [14:44:43] ==> INFO: Installed OpenSSH version 7.4p1, upgrade required
BOA [14:44:43] ==> INFO: Building OpenSSH 7.9p1 from sources, please wait...
BOA [14:47:35] ==> INFO: Installing MariaDB...
BOA [14:47:46] ==> INFO: Retrieving F1656F24C74CD1D8 key..
BOA [14:47:50] ==> INFO: Retrieving 9334A25F8507EFA5 key..
BOA [14:47:53] ==> INFO: Installing MariaDB 10.1.38 in Debian/stretch
BOA [14:48:48] ==> INFO: Running aptitude full-upgrade...
BOA [14:50:22] ==> INFO: Installed Nginx version nginx/1.10.3, upgrade required
BOA [14:50:22] ==> INFO: Installing Nginx...
BOA [14:53:39] ==> INFO: Installing MySecureShell master-20-03-2015...
BOA [14:55:23] ==> INFO: Installing wkhtmltopdf and wkhtmltoimage x86_64...
BOA [14:55:26] ==> INFO: Fix #1 for libs in Debian stretch
BOA [14:55:26] ==> INFO: Fix #2 for libs in Debian stretch
BOA [14:55:26] ==> INFO: Installing VnStat monitor...
BOA [14:55:43] ==> INFO: Installing a few more tools...   
BOA [14:55:43] ==> INFO: PHP 7.2.16 will be installed now
BOA [14:55:43] ==> INFO: Building PHP 7.2.16 from sources, please wait...     
BOA [14:55:48] ==> INFO: Building PHP 7.2.16 part 1/3
BOA [14:55:53] ==> INFO: Building PHP 7.2.16 part 2/3          
BOA [14:56:53] ==> INFO: Building PHP 7.2.16 part 3/3
BOA [15:12:42] ==> INFO: Installing PhpRedis for PHP 7.2.16...
BOA [15:13:03] ==> INFO: Installing UploadProgress for PHP 7.2.16...
BOA [15:13:10] ==> INFO: Installing JSMin for PHP 7.2.16...
BOA [15:13:19] ==> INFO: Installing Imagick for PHP 7.2.16...
BOA [15:13:37] ==> INFO: Installing LibYAML for PHP...
BOA [15:14:00] ==> INFO: Installing YAML for PHP 7.2.16...
BOA [15:17:11] ==> INFO: Installing Limited Shell 0.9.18.9...
BOA [15:17:25] ==> INFO: Installing Redis update for Debian/stretch...
BOA [15:18:34] ==> INFO: Generating random password for MariaDB
BOA [15:18:34] ==> INFO: Updating MariaDB configuration
BOA [15:18:34] ==> INFO: Restarting MariaDB server...
BOA [15:18:46] ==> INFO: Installing Aegir Master Instance, please wait...
BOA [15:18:47] ==> INFO: MariaDB is listening on 212.111.40.127.
BOA [15:20:10] ==> INFO: Generating DH parameters, 2048 bit, please wait...
BOA [15:21:53] ==> INFO: Aegir Master Instance installed
BOA [15:21:53] ==> INFO: Installing default SSL Wildcard Nginx Proxy...
BOA [15:21:54] ==> INFO: Building Pure-FTPd server from sources, please wait...
BOA [15:21:54] ==> INFO: Generating DH parameters, 2048 bit, please wait...

BOA [15:22:50] ==> INFO: Installing Adminer Manager...
BOA [15:22:50] ==> INFO: Adminer Manager installed
BOA [15:22:50] ==> INFO: Updating init scripts
BOA [15:22:51] ==> INFO: InnoDB log file will be set to 256M...
BOA [15:23:07] ==> INFO: Starting Redis, PHP-FPM and Nginx
BOA [15:23:15] ==> INFO: MariaDB final setup
BOA [15:23:15] ==> INFO: Restarting MariaDB server...
BOA [15:23:26] ==> INFO: MariaDB setup completed
BOA [15:23:26] ==> INFO: You can now log in as root by typing just 'mysql'
BOA [15:23:26] ==> INFO: Cleaning up system swap, it may take a moment...

BOA [15:23:32] ==> INFO: Congratulations, Aegir have been installed successfully!
BOA [15:23:32] ==> NOTE! Please wait 2 min before visiting Aegir at:

BOA [15:23:32] ==> LINK: http://master.server.mydomain.net/user/reset/1/1555946356/2TIMipTJw7tcajgB4nj4pzoULGnTefQCgBj6ewAzxVU/login

BOA [15:23:43] ==> INFO: Installing csf/lfd firewall...
BOA [15:23:46] ==> TEST: csf/lfd firewall should work fine on this system
BOA [15:25:16] ==> INFO: csf/lfd firewall installed
BOA [15:25:16] ==> INFO: Upgrading csf/lfd firewall...
BOA [15:25:18] ==> TEST: csf/lfd firewall should work fine on this system
BOA [15:26:48] ==> INFO: csf/lfd firewall upgrade completed
BOA [15:26:48] ==> INFO: Installing DNS cache pdnsd server...
hkBOA [15:27:14] ==> INFO: Building pdnsd 1.2.9a from sources, please wait...
curl: (6) Could not resolve host: getcomposer.org
mv: cannot stat 'composer.phar': No such file or directory
BOA [15:28:05] ==> CARD: Now charging your credit card for this magic show...
BOA [15:28:08] ==> CARD: It will take a moment to process your payment...
BOA [15:28:14] ==> JOKE: Just kidding !!! Enjoy your Aegir Hosting System :)

BOA [15:28:18] ==> Final post-install cleaning, one moment...
BOA [15:28:29] ==> BYE!
files.aegir.cc: forward host lookup failed: Host name lookup failure : Resource temporarily unavailable
INFO: The mirror files.aegir.cc doesn't respond, let's try default

BOA [15:28:34] ==> Skynet Agent v.BOA-4.0.0-prod welcomes you aboard!

gitlab.com: forward host lookup failed: Host name lookup failure : Resource temporarily unavailable
BOA [15:28:41] ==> WARN: The GitLab mirror repository server doesn't respond...
BOA [15:28:41] ==> INFO: Creating your /root/.c1.octopus.cnf config file
BOA [15:28:42] ==> New Octopus Setup on cws01.cheekucreations.net in progress...
BOA [15:32:03] ==> ERROR: o1.server.mydomain.net doesn't point to your IP: 127.0.0.1
BOA [15:32:03] ==> Please make sure you have a valid A record in your DNS
BOA [15:32:03] ==> It is also possible that DNS change didn't propagate yet
BOA [15:32:03] ==> Bye

BOA in-head completed
Bye
`
I then run the following to check resolution:

`root@li761-127:~# hostname -f`
`hostname: Temporary failure in name resolution`
`root@li761-127:~# cat /etc/network/interfaces`
`# This file describes the network interfaces available on your system
# and how to activate them. For more information, see interfaces(5).

source /etc/network/interfaces.d/*

# The loopback network interface
auto lo
iface lo inet loopback

# The primary network interface
allow-hotplug eth0
iface eth0 inet dhcp

`root@li761-127:~# cat /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
nameserver 127.0.0.1
`
Any clues?

**Comment by EdNett** [comment]
Hi,

I *think* this is the right issue to post in - we have discovered that pdnsd isn't running on any boa head instance that has been updated in the last month. The problem is with the /usr/etc/pdnsd.conf file - for us on line 29 in the first server {  area - just after the two IP addresses of our server providor (online.net) there was a bad entry "ip: online.net" instead of an actual IP address - and this caused pdnsd to fail and fail to restart.  Hope this helps.

Ed

**Comment by namontague** [comment]
I have since been able to successfully run the install/update scripts.

In order to stop the DNS lookup failing
```
INIT A: FATAL ERROR: DNS looks broken for server.domain.com)
```
I had to do the following:

-create a file : /root/.use.local.nameservers.cnf
-copy  /etc/resolv.conf to /var/backups/resolv.conf.vanilla 

This was required to maintain AWS DNS settings during the install/update. 

This is documented here:
https://github.com/omega8cc/boa/blob/master/docs/ctrl/system.ctrl



However this did not fix these other 2 issues:
```
curl: (6) Could not resolve host: getcomposer.org
mv: cannot stat 'composer.phar': No such file or directory

netcat: getaddrinfo: Temporary failure in name resolution
INFO: The mirror files.aegir.cc doesn't respond, let's try default
```

**Comment by petrowsky** [comment]
This is definitely an issue when pdnsd is initially built. If your starting resolv.conf can't be parsed and the proper control file isn't there, then it looks like the DNS gets hosed and the box can't make outbound connections to finish the install.

Here's what I hit. https://github.com/omega8cc/boa/issues/1374#issuecomment-488205821

I had to simply modify pdnsd.conf to have the right dns servers for outbound connections and then finish the install of the o1 satellite using the following.

```
boa in-octopus email@address.com o1 {mini|max|none} head
```

Note, if running on Linode, it looks like you need to stop their auto-configure networking

https://www.linode.com/docs/platform/network-helper/#single-per-linode

**Comment by omega8cc** [comment]
We haven't tested anything on AWS for a long time. Not sure if the problem here is related to pdnsd (now no longer installed by default), or the system IP detection we have switched to doesn't work for AWS networking. Please try BOA head again on a fresh instance and let us know if not installing pdnsd by default helped. Thank you.

**Comment by EdNett** [comment]
Hello,

We'd like to try this, but we can't install on a new instance from scaleway. What "services" must we disable, please? I have already purged mosh, iptables and exim4. I don't know what else to purge?

ERROR: BOA requires minimal, supported OS, with no services installed
ERROR: The only acceptable exceptions are: sshd and mail servers
Bye
root@server1:~# service --status-all
 [ - ]  bootlogd
 [ + ]  cgmanager
 [ - ]  cgproxy
 [ + ]  cron
 [ + ]  dbus
 [ - ]  exim4
 [ + ]  haveged
 [ - ]  hwclock.sh
 [ + ]  irqbalance
 [ + ]  kmod
 [ + ]  networking
 [ + ]  ntp
 [ + ]  procps
 [ - ]  rsync
 [ + ]  rsyslog
 [ - ]  screen-cleanup
 [ + ]  ssh
 [ - ]  stop-bootlogd
 [ - ]  stop-bootlogd-single
 [ - ]  sudo
 [ + ]  sysstat
 [ + ]  udev
 [ + ]  unattended-upgrades
 [ - ]  uuidd

There is no etc/apache2 directory. There is as etc/mysql directory but mysql is not installed.  What keeps causing the "minimal" services error message in a current/modern stretch provisioned server. I hate to waste time this way looking for things that probably have no effect at all in BOA. Can you please review your pre-install checks  that keep giving this error?

Best,

Ed

Ed

**Comment by omega8cc** [comment]
Here the list of directories which shouldn't exist on a clean basic Debian OS which will block the installation:

```
system_check_clean() {
  if [ -e "/etc/nginx" ] \
    || [ -e "/var/aegir/.drush/hostmaster.alias.drushrc.php" ] \
    || [ -e "/etc/apache2" ] \
    || [ -e "/etc/mysql" ] \
    || [ -e "/var/lib/mysql" ]; then
    echo "ERROR: BOA requires minimal, supported OS, with no services installed"
    echo "ERROR: The only acceptable exceptions are: sshd and mail servers"
    echo "Bye"
    clean_pid_exit
  fi
}

```

**Comment by omega8cc** [comment]
We have just improved the error message:

```
system_check_clean() {
  if [ -e "/etc/nginx" ] \
    || [ -e "/var/aegir/.drush/hostmaster.alias.drushrc.php" ] \
    || [ -e "/etc/apache2" ] \
    || [ -e "/etc/mysql" ] \
    || [ -e "/var/lib/mysql" ]; then
    echo "ERROR: BOA installation procedure requires minimal supported OS."
    echo "ERROR: Please make sure you don't have MySQL nor Apache installed."
    echo "ERROR: Here's the list of directories which shouldn't exist:"
    echo "ERROR: /etc/nginx /etc/apache2 /etc/mysql /var/lib/mysql /var/aegir"
    echo "Bye"
    clean_pid_exit
  fi
}

```

**Comment by omega8cc** [comment]
Now with pdnsd not installed by default since BOA-4.0.1 the problem is hopefully gone. Feel free to re-open if there are still any host specific issues we have missed. Thanks @All for the feedback!

**Comment by EdNett** [comment]
First of all thank you for the info and much-improved screen notices about the mysql and/or apache2 empty directories that can stop a boa install.

We have just ran the initial install then a few barracuda up-stables after a reboot, and the only problems we have are similar to ones we had before involving the master instance.  The initial warnings were for failure of nginx to restart and gave the yellow warning error messages for everything: localhost, server, hostmaster (twice) and master instance (4 times).

Then we ran barracuda up-stable again(modifying the barracuda.cnf file to add php7.3 to the original 7.2 install, and add headers, nxasi strong passwords, sr7, BDD, FMG and IMG), and everything but the original hostmaster verified green (localhost, server, Aegir hostmaster 001 and the master instance), then 10 min later both the 
hostmaster and the Aegir hostmaster 001 did not verify - red error:

The command could not be executed successfully (returned: Could not open input file: /var/aegir/drush/drush.php , code: 1)

Then we rebooted and ran barracuda up-stable again, and localhost, master, Aegir hostmaster 002 and server verified green, but then a few min later, Aegir hostmaster 002 and Aegir hostmaster 001 and hostmaster did not verify. red error: - same error.

It appears that the master instance isn't running cron - the same problem we had before.

Hope this helps.
Other than this, everything seems to be working fine. Thank you for the upgrade!
Ed
